src/tiled/offsetmapdialog.cpp

```diff
@@ -80,6 +80,30 @@ QRect OffsetMapDialog::affectedBoundingRect() const
     switch (boundsSelection()) {
     case WholeMap:
         boundingRect = QRect(QPoint(0, 0), mMapDocument->map()->size());
+
+        if (mMapDocument->map()->infinite()) {
+            LayerIterator iterator(mMapDocument->map());
+
+            switch (layerSelection()) {
+            case AllVisibleLayers:
```

Comment:
I'm not sure if taking into account the bounds of the layers is any use at all here. It is at least totally unpredictable from the user's point of view, which makes it basically useless. Probably the "Wrap" checkboxes should be disabled for "Whole Map" when the map is infinite, and the bounds can just be the united bounds of all layers in all cases, adjusted by the offset to make sure no tiles are lost.